### PR TITLE
feat(alerting): Add gotify provider

### PR DESCRIPTION
**Pull request by bugrakocabay**
Hi @TwiN 

## Summary
Regarding the issue #603, added the support for Gotify provider with tests and documentation.

## Checklist

- [X] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [X] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

File: alerting/provider/gotify/gotify.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+const DefaultPriority = 5
+
+// AlertProvider is the configuration necessary for sending an alert using Gotify
+type AlertProvider struct {
+	// ServerURL is the URL of the Gotify server
+	ServerURL string `yaml:"server-url"`
+
+	// AppToken is the token of the application to send messages to
+	AppToken string `yaml:"app-token"`
```
Comment (TwiN) [inline alerting/provider/gotify/gotify.go:23]: Any chance you could rename this to just token?

**Review by TwiN — CHANGES_REQUESTED** [review]
One small request, but aside from that, excellent work!

File: alerting/provider/gotify/gotify.go (hunk 1 above)
Comment (bugrakocabay) [reply alerting/provider/gotify/gotify.go]: Hi @TwiN ,

Done.

**Review by TwiN — APPROVED** [review]
Excellent work, @bugrakocabay.

**Pull request merged**